### PR TITLE
Switch login page from bloc to riverpod

### PR DESCRIPTION
**Pull request by MahmoudHafezDarwish**

### CODE REVIEW

File: lib/features/auth/presentation/login/login_screen_presenter.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'package:bond/utils/validator.dart';
+import 'package:flutter/foundation.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+final loginScreenProvider = ChangeNotifierProvider.autoDispose<LoginScreenPresenter>(
+  (ref) => LoginScreenPresenter(),
+);
+
+class LoginScreenPresenter extends ChangeNotifier {
+  bool? _isEmailValid;
```
Comment (salahamassi) [inline lib/features/auth/presentation/login/login_screen_presenter.dart:10]: default value false

File: lib/features/auth/presentation/login/login_page.dart [hunk 2]
```diff
@@ -2,78 +2,93 @@ import 'package:auto_route/auto_route.dart';
...
+                  AppButton(
+                    loading: isLoading.value,
+                    onPressed: () async{
+                      loginScreenPresenter.onSignInPressed(
+                        emailController.text.trim(),
+                        passwordController.text.trim(),
+                      );
+                      if ((loginScreenPresenter.isEmailValid ?? false) &&
+                          (loginScreenPresenter.isPasswordValid ?? false)) {
+                        isLoading.value = true;
```
Comment (salahamassi) [inline lib/features/auth/presentation/login/login_page.dart:80]: move this logic to loginScreenProvider

File: lib/features/auth/presentation/login/login_request_provider.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import 'package:bond/features/auth/auth.dart';
+import 'package:bond/features/auth/data/datasource/auth_http_client.dart';
+import 'package:bond/features/auth/data/dto/user_dto.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:one_studio_core/core.dart';
+
+final loginRequestProvider = StateNotifierProvider<LoginRequestProvider,
```
Comment (salahamassi) [inline lib/features/auth/presentation/login/login_request_provider.dart:7]: this must be Future provider

File: lib/utils/validator.dart [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class Validator {
```
Comment (salahamassi) [inline lib/utils/validator.dart:1]: move this file to core/utils

File: lib/core/widgets/bond_password_text_form_field.dart [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+class BondPasswordTextFormField extends HookWidget {
```
[the remark this answers is not in the document]
Comment (salahamassi) [reply lib/core/widgets/bond_password_text_form_field.dart:4]: export it on app widget files

File: lib/features/auth/presentation/login/login_page.dart [hunk 6]
```diff
@@ -2,78 +2,93 @@ import 'package:auto_route/auto_route.dart';
...
-                    const SizedBox(height: 16),
-                    const NewAccountView(),
-                  ],
-                ),
+                  ),
+                  const SizedBox(height: 12),
+                  BondPasswordTextFormField(
+                    controller: passwordController,
+                    hintText: context.localizations.filed_password_label,
+                    errorText: loginScreenPresenter.isPasswordValid ?? true
```
Comment (salahamassi) [inline lib/features/auth/presentation/login/login_page.dart:66]: loginScreenPresenter.passwordErrorMessage

File: lib/core/widgets/bond_password_text_form_field.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (MahmoudHafezDarwish) [reply lib/core/widgets/bond_password_text_form_field.dart:4]: done

File: lib/features/auth/presentation/login/login_screen_presenter.dart (hunk 1 above)
Comment (MahmoudHafezDarwish) [reply lib/features/auth/presentation/login/login_screen_presenter.dart:10]: done

File: lib/features/auth/presentation/login/login_page.dart (hunk 2 above)
Comment (MahmoudHafezDarwish) [reply lib/features/auth/presentation/login/login_page.dart:80]: done

File: lib/utils/validator.dart (hunk 4 above)
Comment (MahmoudHafezDarwish) [reply lib/utils/validator.dart:1]: done

File: lib/features/auth/presentation/login/login_page.dart (hunk 6 above)
Comment (MahmoudHafezDarwish) [reply lib/features/auth/presentation/login/login_page.dart:66]: done

File: lib/features/auth/presentation/login/login_request_provider.dart (hunk 3 above)
Comment (MahmoudHafezDarwish) [reply lib/features/auth/presentation/login/login_request_provider.dart:7]: I tried, but it didn't work out

**Pull request merged**